packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -810,6 +811,7 @@ async function createSummarizer(loader: ILoader, url: string): Promise<ISummariz
 		});
 		// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
 		if (response.status !== 200 || response.mimeType !== "fluid/object") {
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-argument
```

Comment:
I think here we can improve the typing instead of having to further disable linting. The `request()` method was removed from `IContainer` in https://github.com/microsoft/FluidFramework/pull/18520, but it used to [return a `Promise<IResponse>`](https://github.com/microsoft/FluidFramework/pull/18520/files#diff-83cd27418d0926f1ecd1c09752bbb4b6e18f161e3b371fbf4e1ad3c15d925614L419). I think adding `as IResponse` to line 809 would alleviate the issues from `any`. Also leaves me wondering if there should be a check for `response === undefined` somewhere here, and probably a `?.request(...)` when we call it? Not sure if there's risk of line 809 getting a Container that does not have a request method at all, and this just throwing in weird ways. cc @kian-thompson (as author of the PR) for thoughts.

Reply:
Updated with cast. Good call.

Reply:
Looks good to me. The typing is definitely a bit janky here since it's for back-compat.